src/dependency-manager/src/spec/utils/json-schema.ts

```diff
@@ -114,7 +116,7 @@ const mergeDebugSpec = (definitions: Record<string, SchemaObject>): Record<strin
     delete definition.allOf;
     debug_definitions[`${DEBUG_PREFIX}${key}`] = definition;
 
-    if (definition?.properties?.debug && (key === task_spec_name || key === service_spec_name)) {
+    if (definition?.properties?.debug && (key === task_spec_name || key === service_spec_name || key === sidecar_spec_name)) {
```

Comment:
Feels weird to support a debug block for sidecars. Can you undo these changes and remove the debug property on ResourceSpec. Then add an explicit debug property on both Task/ServiceSpec.